codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -187,10 +193,18 @@ public void close(ChannelHandlerContext ctx, ChannelPromise promise) throws Exce
 
     @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
-        ChannelFuture future = ctx.newSucceededFuture();
-        final Collection<Http2Stream> streams = connection().activeStreams();
-        for (Http2Stream s : streams.toArray(new Http2Stream[streams.size()])) {
-            closeStream(s, future);
+        try {
+            ChannelFuture future = ctx.newSucceededFuture();
+            final Collection<Http2Stream> streams = connection().activeStreams();
+            for (Http2Stream s : streams.toArray(new Http2Stream[streams.size()])) {
+                closeStream(s, future);
+            }
+        } finally {
+            try {
+                encoder().close();
+            } finally {
+                decoder().close();
+            }
```

Comment:
@nmittler So this ensures that close is called on both the encoder and decoder. If exceptions are thrown in multiple places, on the last exception will be propagted. The others are "swallowed".